pubsub/gcp/gcp.go

```diff
@@ -133,15 +130,26 @@ var _ pubsub.Publisher = &publisher{}
 var _ pubsub.MultiPublisher = &publisher{}
 
 // NewPublisher will instantiate a new GCP MultiPublisher.
-func NewPublisher(ctx context.Context, projID, topic string, opts ...option.ClientOption) (pubsub.MultiPublisher, error) {
-	client, err := gpubsub.NewClient(ctx, projID, opts...)
+func NewPublisher(ctx context.Context, cfg Config) (pubsub.MultiPublisher, error) {
+	p := &publisher{}
+
+	if cfg.ProjectID == "" {
+		return p, errors.New("project id is required")
+	}
+	if cfg.Topic == "" {
+		return p, errors.New("topic name is required")
+	}
+	if cfg.Subscription == "" {
+		return p, errors.New("subscription name is required")
+	}
+
+	c, err := gpubsub.NewClient(ctx, cfg.ProjectID)
 	if err != nil {
 		return nil, err
 	}
+	p.topic = c.Topic(cfg.Topic)
 
-	return &publisher{
-		topic: client.Topic(topic),
-	}, nil
+	return p, nil
```

Comment:
I prefer to initiate a struct in a single step to make it clear it has all its dependencies. Mind swapping these lines back to the way they were?

Reply:
How do you return here: https://github.com/lopezator/gizmo/blob/c205e43b0c5df6d6c5a9597bee3c1389baa5264c/pubsub/gcp/gcp.go#L137 and here: https://github.com/lopezator/gizmo/blob/c205e43b0c5df6d6c5a9597bee3c1389baa5264c/pubsub/gcp/gcp.go#L140 then? The same is being made in pubsub/aws/aws.go https://github.com/lopezator/gizmo/blob/c205e43b0c5df6d6c5a9597bee3c1389baa5264c/pubsub/aws/aws.go#L34-L71